Run `yarn run quick-build` failed after Mobx use TypeScript 3.0

Welcome to MobX. Please provide as much relevant information as possible!

I have a:

2. [x] Issue:
  * [ ] **Provide error messages including stacktrace**
  * [ ] Provide a **minimal** sample reproduction. **Create a reproduction based on this [sandbox](https://codesandbox.io/s/v3v0my2370)**
  * [x] Did you check this issue wasn't filed before?
  * [x] Elaborate on your issue. What behavior did you expect?

I tried to `yarn run quick-build`, but the compilation failed. 
This command works fine before 16f85861f45687b019949c298efb8d6a3564bd18. It should be caused by the upgrade of TS 3.2.1.

I will try to fix the errors and request a pr. 😆

Here's the error message.
```
src/api/action.ts:71:55 - error TS2345: Argument of type 'IArguments' is not assignable to parameter of type '[any, any, BabelDescriptor]'.
  Type 'IArguments' is missing the following properties from type '[any, any, BabelDescriptor]': 0, 1, 2, pop, and 26 more.

71         return namedActionDecorator(arg2).apply(null, arguments)
                                                         ~~~~~~~~~

src/api/actiondecorator.ts:44:55 - error TS2345: Argument of type 'IArguments' is not assignable to parameter of type '[any, any, any]'.
  Type 'IArguments' is missing the following properties from type '[any, any, any]': 0, 1, 2, pop, and 26 more.

44         return actionFieldDecorator(name).apply(this, arguments)
                                                         ~~~~~~~~~

src/api/observable.ts:78:42 - error TS2345: Argument of type 'IArguments' is not assignable to parameter of type '[Object, string | symbol, (PropertyDescriptor | undefined)?]'.
  Type 'IArguments' is missing the following properties from type '[Object, string | symbol, (PropertyDescriptor | undefined)?]': 0, 1, pop, push, and 25 more.

78         return deepDecorator.apply(null, arguments)
                                            ~~~~~~~~~

src/core/reaction.ts:187:9 - error TS2741: Property '$mobx' is missing in type '() => void' but required in type 'IReactionDisposer'.

187         return r
            ~~~~~~~~

  src/core/reaction.ts:50:5
    50     $mobx: Reaction
           ~~~~~
    '$mobx' is declared here.

src/utils/decorators.ts:105:42 - error TS2345: Argument of type 'IArguments' is not assignable to parameter of type '[DecoratorTarget, string, BabelDescriptor | undefined, any?]'.
  Type 'IArguments' is missing the following properties from type '[DecoratorTarget, string, BabelDescriptor | undefined, any?]': 0, 1, 2, pop, and 26 more.

105             return decorator.apply(null, arguments)
                                             ~~~~~~~~~


Found 5 errors.
```

  * [x] State the versions of MobX and relevant libraries. Which browser / node / ... version?
 - Node - 10.14.1
 - macOS - 10.14 (18A391)

--- a/src/api/action.ts
+++ b/src/api/action.ts
@@ -40,7 +40,11 @@ export interface IActionFactory {
     <T extends Function>(name: string, fn: T): T & IAction
 
     // named decorator
-    (customName: string): (target: Object, key: string | symbol, baseDescriptor?: PropertyDescriptor) => void
+    (customName: string): (
+        target: Object,
+        key: string | symbol,
+        baseDescriptor?: PropertyDescriptor
+    ) => void
 
     // unnamed decorator
     (target: Object, propertyKey: string | symbol, descriptor?: PropertyDescriptor): void
@@ -64,7 +68,7 @@ export var action: IActionFactory = function action(arg1, arg2?, arg3?, arg4?): 
         // apply to instance immediately
         addHiddenProp(arg1, arg2, createAction(arg1.name || arg2, arg3.value))
     } else {
-        return namedActionDecorator(arg2).apply(null, arguments)
+        return namedActionDecorator(arg2).apply(null, arguments as any)
     }
 } as any
 

--- a/src/api/actiondecorator.ts
+++ b/src/api/actiondecorator.ts
@@ -41,7 +41,7 @@ export function namedActionDecorator(name: string) {
             }
         }
         // bound instance methods
-        return actionFieldDecorator(name).apply(this, arguments)
+        return actionFieldDecorator(name).apply(this, arguments as any)
     }
 }
 

--- a/src/api/observable.ts
+++ b/src/api/observable.ts
@@ -75,7 +75,7 @@ function getEnhancerFromOptions(options: CreateObservableOptions): IEnhancer<any
 function createObservable(v: any, arg2?: any, arg3?: any) {
     // @observable someProp;
     if (typeof arguments[1] === "string") {
-        return deepDecorator.apply(null, arguments)
+        return deepDecorator.apply(null, arguments as any)
     }
 
     // it is an observable already, done

--- a/src/core/reaction.ts
+++ b/src/core/reaction.ts
@@ -182,7 +182,7 @@ export class Reaction implements IDerivation, IReactionPublic {
     }
 
     getDisposer(): IReactionDisposer {
-        const r = this.dispose.bind(this)
+        const r = this.dispose.bind(this) as IReactionDisposer
         r[$mobx] = this
         return r
     }

--- a/src/utils/decorators.ts
+++ b/src/utils/decorators.ts
@@ -102,7 +102,7 @@ export function createPropDecorator(
         if (quacksLikeADecorator(arguments)) {
             // @decorator
             decoratorArguments = EMPTY_ARRAY
-            return decorator.apply(null, arguments)
+            return decorator.apply(null, arguments as any)
         } else {
             // @decorator(args)
             decoratorArguments = Array.prototype.slice.call(arguments)

